### ISSUE TITLE
Some LSP Code actions (Hints) are not applied.

### ISSUE THREAD

**Issue by 0rphee** [haskell/haskell-language-server#3547]
### Your environment

Which OS do you use?
MacOS aarch64

Which version of GHC do you use and how did you install it?
9.2.7 from ghcup

How is your project built (alternative: link to the project)?
stack

Which LSP client (editor/plugin) do you use?
[Helix](https://github.com/helix-editor/helix/)

Which version of HLS do you use and how did you install it?
1.10.0.0 from ghcup

Have you configured HLS in any way (especially: a `hie.yaml` file)?
No

### Steps to reproduce

1. `hx file.hs`              # open file.hs with helix editor
2. add `func = \a -> a`
3. LSP hint will appear: "Redundant lambda (...)"
4. `space + a` to apply code action `Apply Redundant lambda`

### Expected behaviour

Code action is executed. 

### Actual behaviour
Error is shown in the editor: `Outdated workspace edit for path/to/file.hs` 
(This seems to only apply to "code hints".)

### Debug information

After asking in the helix repo (https://github.com/helix-editor/helix/issues/6543), I received this response:

> I can reproduce this but this seems like an upstream issue.
> 
> The LS standard contains the notion of a document version. Whenever a document is edited a new unique version is created. Any change is associated with a unique id and if that version doesn't match the client version then the edit is invalid and ignored by the client (as happens here).
> 
> Looking at the logs it seems that the haskell language server sends the correct version for the most part but for some events the version is just set to `Number(0)`. Perhaphs they meant to send `null` (in which case the version is ignored)? Either way there is nothing we can do about this on our side.
> 
> It seems that this currently works in vscode because vscode isn't as strict as it could/should be about enforcing these versions currently. This is a known issue with the vscode language client. See for example [microsoft/vscode-languageserver-node#752 (comment)](https://github.com/microsoft/vscode-languageserver-node/issues/752#issuecomment-795566339).

The log: 
~/.cache/helix/helix.log
```
2023-04-01T16:01:07.494 helix_term::commands::lsp [ERROR] outdated workspace edit for "/Users/roger/Documents/Dev/haskell-proj/schedule-maker/src/Lib.hs", expected 3548 but got 0
2023-04-01T16:01:07.494 helix_view::editor [ERROR] editor error: outdated workspace edit for "/Users/roger/Documents/Dev/haskell-proj/schedule-maker/src/Lib.hs"
```

**Comment by July541** [comment]
Can you test if the action for class instance works? I think it will report an error too.

<img width="516" alt="image" src="https://user-images.githubusercontent.com/16057632/232281799-643a899a-ac8b-4d16-bf6e-23833098174d.png">

**Comment by 0rphee** [comment]
Well, at first it worked, then it didn't:

https://user-images.githubusercontent.com/79347623/232635775-0fa6ed50-5409-4477-b29c-b35233c71f4d.mov

The helix log:
```
2023-04-17T17:57:09.373 helix_lsp::transport [ERROR] err <- "2023-04-17T23:57:09.373126Z | Info | class: Detected implemented methods for class \"F\": []\n"
2023-04-17T17:57:10.786 helix_term::commands::lsp [ERROR] outdated workspace edit for "/Users/roger/Documents/Dev/main.hs", expected 67 but got 0
```

**Comment by 0rphee** [comment]
Here's also the example video of the problem I posted in the original helix issue:
https://user-images.githubusercontent.com/79347623/229311308-152ad176-0968-432d-ac9b-454fad396647.mov

**Comment by July541** [comment]
Well, I see it. Indeed we send version 0 to the client and it caused code action to fail.

**Comment by maralorn** [comment]
This should be fixed by the just merged #3643.

**Comment by 0rphee** [comment]
Yeah it works just right now! thanks!
